tracing-subscriber/src/layer/context.rs

```diff
@@ -0,0 +1,473 @@
+    ///
+    /// <div class="example-wrap" style="display:inline-block">
+    /// <pre class="ignore" style="white-space:normal;font:inherit;">
+    /// <strong>Note</strong>: This requires the wrapped subscriber to implement the
+    /// <a href="../registry/trait.LookupSpan.html"><code>LookupSpan</code></a> trait.
+    /// See the documentation on <a href="./struct.Context.html"><code>Context</code>'s
+    /// declaration</a> for details.
+    /// </pre></div>
+    #[inline]
+    pub fn exists(&self, id: &span::Id) -> bool
+    where
+        S: for<'lookup> LookupSpan<'lookup>,
+    {
+        self.subscriber.as_ref().and_then(|s| s.span(id)).is_some()
+    }
+
+    /// Returns [stored data] for the span that the wrapped subscriber considers
+    /// to be the current.
+    ///
+    /// If this returns `None`, then we are not currently within a span.
+    ///
+    /// <div class="example-wrap" style="display:inline-block">
+    /// <pre class="ignore" style="white-space:normal;font:inherit;">
+    /// <strong>Note</strong>: This requires the wrapped subscriber to implement the
+    /// <a href="../registry/trait.LookupSpan.html"><code>LookupSpan</code></a> trait.
+    /// See the documentation on <a href="./struct.Context.html"><code>Context</code>'s
+    /// declaration</a> for details.
+    /// </pre></div>
+    ///
+    /// [stored data]: ../registry/struct.SpanRef.html
+    #[inline]
+    pub fn lookup_current(&self) -> Option<registry::SpanRef<'_, S>>
+    where
+        S: for<'lookup> LookupSpan<'lookup>,
+    {
+        let subscriber = *self.subscriber.as_ref()?;
+        let current = subscriber.current_span();
+        let id = current.id()?;
+        let span = subscriber.span(id);
+        debug_assert!(
+            span.is_some(),
+            "the subscriber should have data for the current span ({:?})!",
+            id,
+        );
+
+        // If we found a span, and our per-layer filter enables it, return that span!
+        if let Some(span) = span?.try_with_filter(self.filter) {
+            return Some(span);
+        }
+
+        // Otherwise, the span at the *top* of the stack is disabled by
+        // per-layer filtering, but there may be additional spans in the stack.
+        //
+        // Currently, `LookupSpan` doesn't have a nice way of exposing access to
+        // the whole span stack. However, if we can downcast the innermost
+        // subscriber to a a `Registry`, we can iterate over its current span
+        // stack.
+        //
+        // TODO(eliza): when https://github.com/tokio-rs/tracing/issues/1459 is
+        // implemented, change this to use that instead...
+        #[cfg(feature = "registry")]
+        return self.lookup_current_filtered(subscriber);
+
+        #[cfg(not(feature = "registry"))]
+        None
+    }
+
+    /// Slow path for when the current span is disabled by PLF and we have a
+    /// registry.
+    #[inline(never)]
```

Comment:
I mean, it's `inline(never)` mainly because the outer method is allowed to be inlined, and this is an entire loop...we don't want to inline the loop body, or the (potentially) large amount of downcasting.

Reply:
ah, thanks. i didn't put two-and-two together. do you mind adding what you just explained as a comment to this, in that case? I _know_ that I will forget it